src/Sets/SparsePolynomialZonotope.jl

```diff
@@ -370,6 +370,41 @@ function remove_redundant_generators(S::SparsePolynomialZonotope)
     return SparsePolynomialZonotope(c, G, GI, E)
 end
 
+function reduce_order(P::SparsePolynomialZonotope, ρ::Real)
+    @assert ρ ≥ 1
+    n = dim(P)
+    h = ngens_dep(P)
+    q = ngens_indep(P)
+
+    c = center(P)
+    G = genmat_dep(P)
+    GI = genmat_indep(P)
+    E = expmat(P)
+    idx = indexvector(P)
+
+    a = max(0, min(h + q, ceil(Int, h + q - n * (ρ - 1))))
+    Gbar = hcat(genmat_dep(P), genmat_indep(P))
+    norms = sort([norm(g) for g in eachcol(Gbar)])
+    th = norms[a]
+
+    # TODO: case a = 0
+    K = [norms[i] ≤ th for i in 1:h] # ? Is constructing an array of booleans the most efficient way?
+    Kbar = .!K
+
+    H = [norms[h+i] ≤ th for i in 1:q]
+    Hbar = .!H
```

Comment:
I do not think this works. Since you sorted `norms`, the indices are not the same anymore. Maybe you want to do this instead:
```suggestion
    norms = [norm(g) for g in eachcol(Gbar)]
    th = sort(norms)[a]

    # TODO: case a = 0
    K = [norms[i] ≤ th for i in 1:h] # ? Is constructing an array of booleans the most efficient way?
    Kbar = .!K

    H = [norms[h+i] ≤ th for i in 1:q]
    Hbar = .!H
```